### ISSUE TITLE
Startup silently fails if WORKDIR is in a VOLUME

### ISSUE THREAD

**Issue by jamshid** [docker/for-mac#1085]
### Expected behavior

This Dockerfile should build and run. It runs and outputs expected lines on docker 1.12.4 on Ubuntu 16. It fails to output anything and exits with status code 0 on Docker for Mac beta (Server Version: 1.13.0-rc4).
Sorry I don't know if this problem is related to Docker for Mac or the docker 1.13 RC.

Dockerfile:
```
FROM centos:6.8
VOLUME /var/log
CMD ["/bin/bash", "-c", "echo hello && echo goodbye"]
WORKDIR /var/log
```

### Information
```
$ docker info
Containers: 41
 Running: 32
 Paused: 0
 Stopped: 9
Images: 915
Server Version: 1.13.0-rc4
Storage Driver: overlay2
 Backing Filesystem: extfs
 Supports d_type: true
 Native Overlay Diff: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins: 
 Volume: local
 Network: bridge host ipvlan macvlan null overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 03e5862ec0d8d3b3f750e19fca3ee367e13c090e
runc version: 51371867a01c467f08af739783b8beafc154c4d7
init version: 949e6fa
Security Options:
 seccomp
  Profile: default
Kernel Version: 4.8.15-moby
Operating System: Alpine Linux v3.4
OSType: linux
Architecture: x86_64
CPUs: 2
Total Memory: 3.855 GiB
Name: moby
ID: IJO3:WGF3:7KTH:7XKL:EILQ:HFEC:WOVQ:3XUD:SLTS:QVNI:OFK3:3TSU
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): true
 File Descriptors: 263
 Goroutines: 214
 System Time: 2016-12-27T05:09:43.219525971Z
 EventsListeners: 1
Registry: https://index.docker.io/v1/
Experimental: true
Insecure Registries:
 192.168.2.61:5000
 127.0.0.0/8
Live Restore Enabled: false
```
9D052C1E-09A7-4126-BC50-6FC7D513ED7C

### Steps to reproduce the behavior

  1. Put above Dockerfile in a directory and run:
```
docker run -ti $(docker build -q .)
```
On failing Docker for Mac beta it outputs nothing and exits. On working Docker on Ubuntu 16 it outputs expected lines. 
```
hello
goodbye
```
Interesting: seems to work if you move the WORKDIR line *above* CMD.

**Comment by justincormack** [comment]
Hi, this is a docker issue see https://github.com/docker/docker/issues/29667

**Comment by justincormack** [comment]
I temporary workaround is to add any command below the WORKDIR

**Comment by justincormack** [comment]
This will be fixed in 1.13.0-rc5 which should be out this week.

**Comment by dsheets** [comment]
This should have been fixed in Beta 35. Beta 36 is current. If you still have this problem, please feel free to re-open this issue.

Thanks for using Docker for Mac!
